src/Tool/ActorProviderInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Doctrine Behavioral Extensions package.
+ * (c) Gediminas Morkevicius <gediminas.morkevicius@gmail.com> http://www.gediminasm.org
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Gedmo\Tool;
+
+/**
+ * Interface for a provider for an actor for extensions supporting actor/user references.
+ */
+interface ActorProviderInterface
+{
+    /**
+     * @return object|string
```

Comment:
should we support returning `null` for the case where we don't have an actor or should we throw ? If throwing, what exception should be used ?

Reply:
BlameableListener supports `null`, so I suggest using `@return object|string|null`

Reply:
Yeah I thought about that the other day, null is definitely needed here and it's updated with that.